### DISCUSSION TITLE
Auto rotate pictures, or/and edit existing photos without creating a copy

### DISCUSSION THREAD

**Discussion by Bentsi2829** [ente-io/ente#2535] [Enhancements]
I have recently switched to Ente, and find that hundreds of my wedding photos are rotated 90 degrees to the side. My guess is that google photos auto-rotate them for me, when i was viewing them in google photos. 

When I then go to rotate them back the right way in Ente, my only options is to eather download the new version (of the picture rotate the right way), or copy the new version to Ente (save a copy to Ente). 
As I see it, there are 2 problems with saving a new copy. 1 I am only interested in the new version of the picture, going back to delete the old one is tedious. 2 When you create a new version, it creates a new date for the new picture. This means that you have to rotate, save a new copy, delete the old version, fix the date of the new version. - and you have to do this for every picture you want to rotate 90 degrees. 

1. There needs to be a way to edit your existing photo, without creating a new one. 
2. A auto-rotate feature built into the software would be sweet. - And would save much time and frustration. 

Thanks for reading

**Comment by fabian19vs** [comment]
Totally agree, I have tons of photos that need to be rotated 90 or 180 degrees. I think that the app could display the photo with the chosen orientation (and save those settings in a separate file) without modifying the original photo.

**Comment by stelioskat** [comment]
Yes, please! Every time I want to rotate, I have also to delete the original. Generally, it would be great to be able to choose, if you want to edit the original or create a copy before exiting the edit mode. Similar to apple photos.

**Comment by felixblaschke** [comment]
This was a blocker to not migrate to ente. Found a lot of old pictures wrongly rotated and i just couldn't fix it. Also it would be cool if face recognition could be used to guess the correct orientation.

**Comment by chrisdwlittle** [comment]
Really enjoying Ente for the most part, but this is an incredibly frustrating gap in the experience compared to Google Photos. It makes using the service much more cumbersome for me. 

Is there any chance this will be implemented?

**Comment by pdmarf** [comment]
same here! please implement

**Comment by jcreixell** [comment]
+1, I would love to have this feature

**Comment by kb336** [comment]
That seems logical... please implement this function 🙌🏻

**Comment by rtmkrlv** [comment]
I recently switched from Google Photos to ente, and many of my photos are displayed in the wrong orientation. In Google Photos, the orientation was detected automatically, which was very convenient. In ente, I can use the editor, but rotating a large number of photos manually takes a long time, and if I do it through the built-in editor, a copy of the photo is created. Overall, the inability to easily rotate the original photo is very inconvenient.

**Comment by zilexa** [comment]
Ah ok just discovered this issue. No auto rotate. And when you do it in Ente it creates a copy. So annoying. 

This means Aves Gallery will stay as my Gallery app. Ente is just there to create a backup.

**Comment by Brouware** [comment]
This problem is even worse than described so far. When I use Ente's rotate feature to fix photo's with an incorrect orientation, it creates a copy and cannot fix the original as others mentioned. But the copy has a completely different naming scheme when doing this in the Android app, and deletes certain metadata. If I took multiple photo's in a single minute, I can no longer see at what second of that minute I took that picture, for instance. When I rotate in the Flatpak Linux app, the name is identical with '-edited' appended. Much better than a random number. But it deletes even more metadata, among which the GPS location. Unacceptable in my opinion, if I didn't realize this and deleted the original, I would have lost valuable data. Also, the copies are compressed so there is also quality lost. I think it would be better not to offer this feature at all than in the current state.

**Comment by zilexa** [comment]
I've already decided, Ente is just a backup tool, until my subscription ends.They keep focusing on all these "social" features, because an analyst told them those would generate more revenue (gain more subscriptions) but the most basic stuff is not working. Its either that or their engineers suffer from tunnel vision. 

Ialready know 2 people just as disappointed as me who are not going to renew their subscription because of these issues. They felt mislead. 

Main reason: ente claims its a non destructive service. That's misleading! Because very normal, very common actions that users might take will lead to loss of original data. Plenty of examples available above. 

I edit everything via Digikam on my laptop. It's very tedious work to then delete in Ente and upload again. I didn't know that Ente would give me this much work when i started. The documentation is nice but should contain big fat warnings. This feels like Alpha software with a lot of added social schmuck. 

I was so excited when I noticed an update in the Play Store today.. well, they improved Memories. :sigh:

**Reply by Brouware** [reply]
I agree, the basic stuff should get priority. Above all it should be reliable.
